lib/msal-common/src/telemetry/performance/PerformanceClient.ts

```diff
@@ -82,6 +91,76 @@ export abstract class PerformanceClient implements IPerformanceClient {
      */
     abstract generateId(): string;
 
+    /**
+     * Returns current time in milliseconds. 
+     * 
+     * @returns {number}
+     */
+    abstract getCurrentTime(): number;
+
+    /**
+     * Calculates the difference between current time and time when function was queued.
+     * Note: It is possible to have 0 as the queue time if the current time and the queued time was the same.
+     * 
+     * @param {number} preQueueTime 
+     * @param {number} currentTime 
+     * @returns {number}
+     */
+    calculateQueuedTime(preQueueTime: number, currentTime: number): number {        
+        // More number calculations here?
+        if (Number.isInteger(preQueueTime) && preQueueTime < 1) {
+            this.logger.trace(`PerformanceClient: preQueueTime should be a positive integer and not ${preQueueTime}`);
+            return 0;
+        }
+
+        if (Number.isInteger(currentTime) && currentTime < 1) {
+            this.logger.trace(`PerformanceClient: currentTime should be a positive integer and not ${currentTime}`);
+            return 0;
+        }
+
+        if (currentTime < preQueueTime) {
+            this.logger.trace("PerformanceClient: currentTime is less than preQueueTime, check how time is being retrieved");
+            return 0;
+        }
+
+        return currentTime-preQueueTime;
+    }
+
+    /**
+     * Adds queue measurement time to QueueMeasurements array for given correlation ID.
+     * 
+     * @param {PerformanceEvents} name 
+     * @param {?string} correlationId 
+     * @param {?number} time 
+     * @returns 
+     */
+    addQueueMeasurement(name: PerformanceEvents, correlationId?: string, time?: number): void {
+        if (!correlationId) {
```

Comment:
I looked into this, but did not make correlation Id required for the time being, because there are times where we are adding queue measurements before our APIs are generating correlation Id as one of the first steps.